tests/puzzle/test_provider.py

```diff
@@ -95,6 +95,63 @@ def test_search_for_vcs_read_setup(provider, mocker):
     }
 
 
+@pytest.mark.skipif(not PY35, reason="AST parsing does not work for Python <3.4")
+def test_search_for_vcs_read_setup_with_single_tag(provider, mocker):
+    mocker.patch("poetry.utils.env.Env.get", return_value=MockEnv())
+
+    dependency = VCSDependency(
+        "demo", "git", "https://github.com/demo/demo.git", tag="0.1.2"
+    )
+    assert dependency.branch is None
+    assert dependency.rev is None
+    assert dependency.tag == "0.1.2"
+
+    packages = provider.search_for_vcs(dependency)
+    assert len(packages) == 1
+    package = packages[0]
+    assert package.name == "demo"
+    assert package.version.text == dependency.tag
+
+
+@pytest.mark.skipif(not PY35, reason="AST parsing does not work for Python <3.4")
+def test_search_for_vcs_read_setup_with_missing_tag(provider, mocker):
+    mocker.patch("poetry.utils.env.Env.get", return_value=MockEnv())
+
+    dependency = VCSDependency(
+        "demo", "git", "https://github.com/demo/demo.git", tag="0.2.0"
+    )
+    assert dependency.branch is None
+    assert dependency.rev is None
+    assert dependency.tag == "0.2.0"
+
+    packages = provider.search_for_vcs(dependency)
+    assert len(packages) == 1
+    package = packages[0]
+    assert package.name == "demo"
+    assert package.version.text != dependency.tag
+    assert package.version.text == "0.1.2"
```

Comment:
This is the current behavior, captured in a test.  Not clear yet how/why it gets "0.1.2" here (must be in mocks somewhere).